### PR TITLE
No version number in PM2 - Closes #538

### LINKED ISSUE #538: No version number in PM2
Would be nice to see the version:

image

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?

This PR resolves #538

### How was it solved?
- [x] Add configuration for local and remote deployment using pm2

### How was it tested?
- [x] Manually tested on local machine and remote server

__Steps to test:__

*_LOCAL DEPLOYMENT_*

* Turn on the remote connection over ssh
* Install pm2, python2
* Move nvm config to top of `.bashrc` or `.zshrc`
* Setup server: `pm2 deploy local setup`
* Deploy: `pm2 deploy local`

*_REMOTE DEPLOYMENT_*

* [Install ssh certificate on the server](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
* Update ecosystem.config.js with correct `host`, `user`, `ref`, `repo`, `path`
* Move nvm config to top of `.bashrc` or `.zshrc`
* Setup server: `pm2 deploy remote setup`
* Deploy: `pm2 deploy remote`

---

__Steps to install Python 2.7.18:__
* *Linux*:
  * `sudo apt install python`
* *MacOS*:
  * `brew install pyenv`
  * `pyenv install 2.7.18`
  * `export PATH="$(pyenv root)/shims:${PATH}"`
  * `echo 'PATH=$(pyenv root)/shims:$PATH' >> ~/.zshrc`
  * `pyenv global 2.7.18`

### CODE REVIEW

File: ecosystem.config.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * LiskHQ/lisk-service
```
Comment (ManuGowda) [inline ecosystem.config.js:2]:
```suggestion
```

**Review by ManuGowda — APPROVED** [review]

File: ecosystem.config.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			script: './bin/run',
+			args: 'start',
+			env: {
+				LISK_NETWORK: 'mainnet',
+				LISK_API_IPC: 'true',
+			},
+			pid_file: './pids/lisk-core.pid',
+			out_file: './logs/lisk-core.log',
+			error_file: './logs/lisk-core.err',
+			log_date_format: 'YYYY-MM-DD HH:mm:ss SSS',
```
Comment (sameersubudhi) [inline ecosystem.config.js:29]: Logging time granularity up to seconds should be fine.
```suggestion
			log_date_format: 'YYYY-MM-DD HH:mm:ss,
```

File: ecosystem.config.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		},
+	],
+	deploy: {
+		local: {
+			user: 'root',
+			host: '127.0.0.1',
+			ref: 'origin/release/4.0.0',
```
Comment (sameersubudhi) [inline ecosystem.config.js:36]: The `ref` should be set to the `main` branch by default.
```suggestion
			ref: 'origin/main',
```

File: ecosystem.config.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			repo: 'https://github.com/LiskHQ/lisk-core.git',
+			path: '/Users/{USER_NAME}/temp_deploy',
+			'pre-setup': 'rm -r /Users/{USER_NAME}/temp_deploy',
+			'post-setup': 'source ~/.bashrc && source ~/.zshrc && nvm install && npm ci && npm run build',
+			'post-deploy': 'source ~/.bashrc && source ~/.zshrc && pm2 start ecosystem.config.js',
+		},
+		remote: {
+			user: 'root',
+			host: ['8.8.8.8'],
+			ref: 'origin/release/4.0.0',
```
Comment (sameersubudhi) [inline ecosystem.config.js:46]: Same as above.
```suggestion
			ref: 'origin/main',
```

File: ecosystem.config.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	deploy: {
+		local: {
+			user: 'root',
+			host: '127.0.0.1',
+			ref: 'origin/release/4.0.0',
+			repo: 'https://github.com/LiskHQ/lisk-core.git',
+			path: '/Users/{USER_NAME}/temp_deploy',
+			'pre-setup': 'rm -r /Users/{USER_NAME}/temp_deploy',
+			'post-setup': 'source ~/.bashrc && source ~/.zshrc && nvm install && npm ci && npm run build',
+			'post-deploy': 'source ~/.bashrc && source ~/.zshrc && pm2 start ecosystem.config.js',
```
Comment (sameersubudhi) [inline ecosystem.config.js:41]: Please verify what happens when either of `~/.bashrc` or `~/.zshrc` doesn't exist and handle the situation accordingly.

File: ecosystem.config.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+		},
+		remote: {
+			user: 'root',
+			host: ['8.8.8.8'],
+			ref: 'origin/release/4.0.0',
+			repo: 'https://github.com/LiskHQ/lisk-core.git',
+			path: '/root/home/temp_deploy',
+			'pre-setup': 'rm -r /root/home/temp_deploy',
+			'post-setup': 'source ~/.bashrc && nvm install && npm ci && npm run build',
+			'post-deploy': 'source ~/.bashrc && pm2 start ecosystem.config.js',
```
Comment (sameersubudhi) [inline ecosystem.config.js:51]: Same as above.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: ecosystem.config.js (hunk 6 above)
Comment (priojeetpriyom) [reply ecosystem.config.js]: Applied

File: ecosystem.config.js (hunk 5 above)
Comment (priojeetpriyom) [reply ecosystem.config.js]: Applied. It did not execute following commands, fixed this buy pipelining with a fallback true condition. Thanks for the suggestion 👍

File: ecosystem.config.js (hunk 4 above)
Comment (priojeetpriyom) [reply ecosystem.config.js]: Applied

File: ecosystem.config.js (hunk 3 above)
Comment (priojeetpriyom) [reply ecosystem.config.js]: Applied.

File: ecosystem.config.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		},
+		remote: {
+			user: 'root',
+			host: ['8.8.8.8'],
+			ref: 'origin/main',
+			repo: 'https://github.com/LiskHQ/lisk-core.git',
+			path: '/root/home/temp_deploy',
+			'pre-setup': 'rm -r /root/home/temp_deploy',
+			'post-setup': 'source ~/.bashrc || true && nvm install && npm ci && npm run build',
+			'post-deploy': 'source ~/.bashrc || true && pm2 start ecosystem.config.js',
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply ecosystem.config.js:53]: checking

File: ecosystem.config.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+module.exports = {
+	apps: [
+		{
+			name: 'lisk-core',
+			script: './bin/run',
+			args: 'start',
+			env: {
+				LISK_NETWORK: 'mainnet',
```
Comment (nagdahimanshu) [inline ecosystem.config.js:23]: I would also recommend to add all the supported environment variables with thier default values
```suggestion
				LISK_NETWORK: 'mainnet',
				LISK_API_IPC: true,
				LISK_ENABLE_FORGER_PLUGIN: true,
				LISK_API_WS: true,
				LISK_API_WS_PORT: 7887,
				LISK_CONSOLE_LOG_LEVEL:  'warn',				
```

File: ecosystem.config.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply ecosystem.config.js]: Applied

File: ecosystem.config.js (hunk 8 above)
Comment (priojeetpriyom) [reply ecosystem.config.js:23]: Applied

**Review by nagdahimanshu — APPROVED** [review]
Looks good to me 👍

**Review by sameersubudhi — APPROVED** [review]
Thank you for the contribution @priojeetpriyom!! 🙏

**Pull request merged**